ecs.tf

```diff
@@ -32,19 +32,35 @@ resource "aws_alb_listener_rule" "listener_rule" {
   condition = [
     {
       field  = "host-header"
-      values = ["${aws_route53_record.dns_record.name}"]
+      values = ["${coalescelist(aws_route53_record.dns_record.*.name, list(var.dns_record_name))}"]
+
     },
     {
       field  = "path-pattern"
-      values = ["${var.alb_listener_path_pattern}"]
+      values = ["${element(var.alb_listener_path_patterns, count.index)}"]
     },
   ]
 }
 
-resource "aws_alb_listener_rule" "listener_rule_existing" {
-  count        = "${var.dns_record_name == "" ? 0 : 1}"
+resource "aws_alb_listener_rule" "listener_rule_auth" {
+  count        = "${var.auth_domain != "" ? length(var.alb_listener_path_patterns) : 0}"
   listener_arn = "${var.aws_alb_listener_arn}"
-  priority     = "${var.listener_rule_priority}"
+  priority     = "${var.listener_rule_priority + count.index}"
+
+  action {
+    type = "authenticate-oidc"
+
+    authenticate_oidc {
+      authorization_endpoint = "${var.auth_domain}authorize"
+      client_id              = "${var.auth_client_id}"
+      client_secret          = "${var.auth_client_secret}"
+      issuer                 = "${var.auth_domain}"
+      token_endpoint         = "${var.auth_domain}oauth/token"
+      user_info_endpoint     = "${var.auth_domain}userinfo"
```

Comment:
Maybe make the entire string a variable?

Reply:
Would you do it for the `token_endpoint` and `authorization_endpoint` above as well? Or do you feel those are generic and `user_info` isn't